cmd/akamai-purger/main_test.go

```diff
@@ -0,0 +1,60 @@
+package notmain
+
+import (
+	"testing"
+	"time"
+)
+
+func TestThroughput_validate(t *testing.T) {
+	type fields struct {
+		ResponsesPerBatch int
+		PurgeBatchEvery   time.Duration
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		wantErr bool
+	}{
+		{"deployability defaults",
+			fields{
+				ResponsesPerBatch: 33,
+				PurgeBatchEvery:   10 * time.Second},
+			false,
+		},
+		{"optimized defaults, should succeed",
+			fields{
+				ResponsesPerBatch: defaultResponsesPerBatch,
+				PurgeBatchEvery:   defaultPurgeBatchEvery},
+			false,
+		},
+		{"2ms faster than optimized defaults, should succeed",
+			fields{
+				ResponsesPerBatch: defaultResponsesPerBatch,
+				PurgeBatchEvery:   30 * time.Millisecond},
+			false,
+		},
+		{"3ms faster than optimized defaults, exceed by 4 URLs",
+			fields{
+				ResponsesPerBatch: defaultResponsesPerBatch,
+				PurgeBatchEvery:   29 * time.Millisecond},
+			true,
+		},
+		{"exceed by 20 bytes",
+			fields{
+				ResponsesPerBatch: 125,
+				PurgeBatchEvery:   1 * time.Second},
+			true,
+		},
+	}
+	for _, tt := range tests {
```

Comment:
These test cases should clearly differentiate between the ones exercising the "too many requests per second" limit and the ones exercising the "too many URLs per second". (I'm not sure I see any which exercise the former?)